src/plugins/input_control_vis/public/vis_controller.tsx

```diff
@@ -50,19 +54,30 @@ export const createInputControlVisController = (deps: InputControlVisDependencie
       this.updateSubsciption = this.filterManager
         .getUpdates$()
         .subscribe(this.queryBarUpdateHandler);
+      this.timeFilterSubscription = deps.data.query.timefilter.timefilter
+        .getTimeUpdate$()
+        .subscribe(() => {
+          if (this.visParams?.useTimeFilter) {
+            this.isLoaded = false;
+          }
+        });
```

Comment:
I'm not sure do you really need the subscription? 
I didn't find the case where do you need to handle this specific parameter.
Could you please clarify this approach? Thnx

Reply:
This subscription is used to re-initialize the input controls when the 'useTimeFilter' setting is on, and the time filter has changed. This is an expected behaviour of input controls, and was missing after I stopped the 're-initialize everything on each render' behavior.